### PR TITLE
Add `div_by_2()` to `Monty` trait

### PR DESCRIPTION
**Pull request by fjarri**
- add `Monty::div_by_2()` method
- make `MontyForm::div_by_2()` and `ConstMontyForm::div_by_2()` const 
- make the internal `div_by_2()` take an `Odd`-wrapped modulus

Thought about making it a trait (like `Square`), but it's somewhat tied to the Montgomery modulus.

### CODE REVIEW

**Comment by fjarri** [comment]
Actually wait, let me rephrase the docstrings. They're currently a little confusing.

**Review by tarcieri — APPROVED** [review]
